core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -2710,6 +2710,16 @@ private boolean accept1(RexNode e, SqlKind kind,
       }
     }
 
+    /** Checks whether it is worth to fix and convert to {@code SEARCH} calls. */
+    boolean needToFix(int newTermsCnt) {
+      // Fix and converts to SEARCH if:
+      // 1. A Sarg has complexity greater than 1;
+      // 2. The terms are reduced as simpler Sarg points.
+      return map.values().stream().anyMatch(b -> b.complexity() > 1)
```

Comment:
Streams are cool but they are very expensive.
Probably we are in some hot path here.
Is it worth to use a simple loop?

Reply:
+ 1 for this suggestion.
By using a simple loop, we can combine the two stream expressions?
In addition, checking condition `newTermsCnt == 1` is cheap, can we move it forward?

Reply:
Sorry i found that it is hard to keep the decision branches in just one for loop and make the logic clear and clean. So i would not follow that.